### ISSUE TITLE
[question] Is it possible to install a package by package ID only?

### ISSUE THREAD

**Issue by unzap** [conan-io/conan#4926]
The local cache/server contains e.g. the following artifacts:

Existing packages for recipe foo/1.2.0@bar/channel:

`    Package_ID: 254f204bed49cb02b180703c0d196c9394f33bd6`
`        [settings]`
`            setting_1: aaa`
`            setting_2: bbb`
`            setting_3: ccc`
`            setting_4: ddd`
`            setting_5: eee`
`        Outdated from recipe: False`

`    Package_ID: 8537c5a6ddfc03df794fb5128b69464281b79bda`
`        [settings]`
`            setting_1: qqq`
`            setting_2: www`
`            setting_3: eee`
`            setting_4: rrr`
`            setting_5: ttt`
`        Outdated from recipe: False`

Now I need to install a package like this:

`conan install foo/1.2.0@bar/channel -s setting_1=aaa -s setting_2=bbb -s setting_3=ccc -s setting_4=ddd -s setting_5=eee` 

If I try to install with the Package_ID:

`conan install foo/1.2.0@bar/channel#8537c5a6ddfc03df794fb5128b69464281b79bda`

it gives:

`ERROR: foo/1.2.0@bar/channel: 'settings.aaa' value not defined`

Is there any way for a "shorthand"?

**Comment by lasote** [comment]
No, Conan doesn't work like that. You need the settings to determine the hash, and as it is a hash, you cannot get the settings from it.
You could download a concrete package_id with `conan download` command, but for using it as a dependency you need to input the settings to resolve the whole graph.
